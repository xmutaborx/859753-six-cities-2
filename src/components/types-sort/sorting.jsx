import React from 'react';
import PropTypes from 'prop-types';

import {SORT_OPTION} from '../../constants/constants';
import withSorting from '../../hocs/with-sorting/with-sorting.jsx';

const Sorting = (props) => {
  const {onToggleList, onChangeType, isOpen, sortTypeLabel, sortType} = props;
  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span className="places__sorting-type" tabIndex="0" onClick={onToggleList}>
        {sortTypeLabel}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select" />
        </svg>
      </span>
      <ul className={`places__options places__options--custom ${isOpen ? `places__options--opened` : ``}`}>
        {SORT_OPTION.map((it) => (
          <li
            className={`places__option ${sortType === it.type ? `places__option--active` : ``}`}
            tabIndex="0"
            onClick={() => onChangeType(it)}
            key={it.type}
          >{it.value}</li>
        ))}
      </ul>
    </form>
  );
};

Sorting.propTypes = {
  onToggleList: PropTypes.func.isRequired,
  onChangeType: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  sortTypeLabel: PropTypes.string.isRequired,
  sortType: PropTypes.string.isRequired,
};

export default withSorting(Sorting);
