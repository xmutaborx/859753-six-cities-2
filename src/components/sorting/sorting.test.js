import React from 'react';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16/build";
import toJson from "enzyme-to-json";

import {Sorting} from './sorting.jsx';

Enzyme.configure({adapter: new Adapter()});

it(`Sorting correctly renders`, () => {
  const tree = shallow(<Sorting
    onToggleList={() => {}}
    onChangeType={() => {}}
    isOpen={false}
    sortTypeLabel={`Popular`}
    sortType={`Popular`}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
