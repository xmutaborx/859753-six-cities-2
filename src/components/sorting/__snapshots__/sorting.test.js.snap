// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting correctly renders 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <li
      className="places__option "
      key="popular"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      key="lowToHigh"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      key="highToLow"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      key="rated"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
